spec/requests/products/show_spec.rb

```diff
@@ -0,0 +1,61 @@
+require "web_spec"
+
+RSpec.describe "GET /products/:id", type: :request do
+  context "when product with given id does not exist" do
+    before do
+      do_request(0)
```

Comment:
can you say what is the `0` for? :P It should be either a hash with `id` key or keyword argument with default value: https://robots.thoughtbot.com/ruby-2-keyword-arguments